apps/trustlab/src/payload/collections/Users/hooks/protectRoleField.js

```diff
@@ -0,0 +1,52 @@
+import {
+  ROLE_ADMIN,
+  ROLE_AUTHOR,
+  ROLE_EDITOR,
+} from "@/trustlab/payload/access/roles";
+
+// Field hook that only runs on the role field
+export const protectRoleField = async ({
+  req,
+  data,
+  originalDoc,
+  operation,
+}) => {
+  // If role is not changed, return original role
+  if (data?.role === originalDoc?.role) {
+    return originalDoc?.role;
+  }
+
+  // Ensure first user is admin
+  if (operation === "create") {
+    const { docs } = await req.payload.find({
+      collection: "users",
+      limit: 1,
+    });
+    if (!docs.length) {
+      return ROLE_ADMIN;
+    }
+  }
+
+  const { user } = req;
+  if (!user) {
+    return originalDoc?.role || ROLE_AUTHOR;
+  }
+
+  switch (user.role) {
+    case ROLE_ADMIN:
+      return data?.role;
+    case ROLE_EDITOR:
+      // Editor can only set role to EDITOR or AUTHOR
+      if ([ROLE_EDITOR, ROLE_AUTHOR].includes(data?.role)) {
+        return data?.role;
+      }
+      // Ensure editors cannot create admin users
+      return operation === "create" ? ROLE_AUTHOR : originalDoc?.role;
```

Comment:
Add a issue to track this.. I assume the system will silently convert say ROLE_EDITOR to ROLE_AUTHOR without warning editors that they can't create editors.